### ISSUE TITLE
Inference with MCTS

### ISSUE THREAD

**Issue by garipovroma** [AIDC-AI/Marco-o1#21]
Hello, thank you for your work and contributions to the community.

I’ve recently developed an interest in your work and took a closer look at your code. However, I couldn’t find the code for inference with MCTS. Could you kindly upload it if possible?

**Comment by zwq2018** [comment]
yeah, very confused. It seems that there is no MCTS at all in the inference stage.

**Comment by Blacknana** [comment]
I have the same confusion. Could you please give an example of MCTS inference? And I'm interested in integrating it into open-source LLM serving like SGLang or vLLM.

**Comment by cjx-1234** [comment]
i have found a wonderful implementation for inference with MCTS from [AlphaMath](https://github.com/MARIO-Math-Reasoning/Super_MARIO) and its [paper](https://arxiv.org/abs/2405.03553). btw, this paper have proposed step-level beam search algorithm which is more efficient and effective than MCTS in inference.
